wagtail/snippets/views/snippets.py

```diff
@@ -1315,13 +1434,143 @@ def get_urlpatterns(self):
 
         return urlpatterns + legacy_redirects
 
+    def get_edit_handler(self):
+        """
+        Returns the appropriate edit handler for this ``SnippetViewSet`` class.
+        It can be defined either on the model itself or on the ``SnippetViewSet``,
+        as the ``edit_handler`` or ``panels`` properties. Falls back to
+        extracting panel / edit handler definitions from the model class.
+        """
+        if hasattr(self, "edit_handler"):
+            edit_handler = self.edit_handler
+        elif hasattr(self, "panels"):
+            panels = self.panels
+            edit_handler = ObjectList(panels)
+        elif hasattr(self.model, "edit_handler"):
+            edit_handler = self.model.edit_handler
+        elif hasattr(self.model, "panels"):
+            panels = self.model.panels
+            edit_handler = ObjectList(panels)
+        else:
+            exclude = self.get_exclude_form_fields()
+            panels = extract_panel_definitions_from_model_class(
+                self.model, exclude=exclude
+            )
+            edit_handler = ObjectList(panels)
+        return edit_handler.bind_to_model(self.model)
+
+    def get_form_class(self, for_update=False):
+        return self._edit_handler.get_form_class()
+
     def register_model_check(self):
         def snippets_model_check(app_configs, **kwargs):
             return check_panels_in_model(self.model, "snippets")
 
         checks.register(snippets_model_check, "panels")
 
+    def register_menu_item(self):
+        if self.add_to_settings_menu:
+            hooks.register("register_settings_menu_item", self.get_menu_item)
+        elif self.add_to_admin_menu:
+            hooks.register("register_admin_menu_item", self.get_menu_item)
+
     def on_register(self):
         super().on_register()
+        # For convenience, attach viewset to the model class to allow accessing
+        # the configuration of a given model.
+        self.model.snippet_viewset = self
         viewsets.register(self.chooser_viewset)
         self.register_model_check()
+        self.register_menu_item()
+
+
+class SnippetViewSetGroup:
+    """
+    A container for grouping together multiple SnippetViewSet instances. Creates
+    a menu item with a submenu for accessing the listing pages of those instances.
+    """
+
+    #: A list or tuple of :class:`SnippetViewSet` classes to be grouped together
+    items = ()
+
+    #: Register a custom menu item for the group in the admin's main menu.
+    add_to_admin_menu = True
+
+    # Undocumented for now, but it is technically possible to register the group's
+    # menu item in the Settings menu instead of the main menu.
+    add_to_settings_menu = False
```

Comment:
I think the Modeladmin docs are saying that setting `add_to_settings_menu = True` on an individual ModelAdmin that's already part of a ModelAdminGroup will have no effect (because the ModelAdminGroup takes precedence on where the item goes), not that you can't put a ModelAdminGroup into the settings menu. Either way - 👍 for more customisability!